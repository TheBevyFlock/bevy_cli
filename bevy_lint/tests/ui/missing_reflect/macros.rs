//@aux-build:../auxiliary/proc_macros.rs
#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::missing_reflect)]
#![allow(dead_code)]

use bevy::prelude::*;
extern crate proc_macros;
use proc_macros::external;

external! {
    #[derive(Component)]
    struct MyComponentExternal;
}

macro_rules! local_macro {
    () => {
        //~ NOTE: `Component` implemented here
        #[derive(Component)]
        //~| HELP: `Reflect` can be automatically derived
        //~v ERROR: defined a component without a `Reflect` implementation
        struct MyComponent;
    };
}

local_macro!();
