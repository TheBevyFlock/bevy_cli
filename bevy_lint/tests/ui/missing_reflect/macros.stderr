error: defined a component without a `Reflect` implementation
  --> tests/ui/missing_reflect/macros.rs:17:9
   |
17 |         struct MyComponent;
   |         ^^^^^^^^^^^^^^^^^^^
...
26 | local_macro!();
   | -------------- in this macro invocation
   |
note: `Component` implemented here
  --> tests/ui/missing_reflect/macros.rs:14:18
   |
14 |         #[derive(Component)]
   |                  ^^^^^^^^^
...
26 | local_macro!();
   | -------------- in this macro invocation
note: the lint level is defined here
  --> tests/ui/missing_reflect/macros.rs:5:9
   |
5  | #![deny(bevy::missing_reflect)]
   |         ^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the macro `local_macro` (in Nightly builds, run with -Z macro-backtrace for more info)
help: `Reflect` can be automatically derived
   |
17 ~         #[derive(Reflect)]
18 ~         struct MyComponent;
   |

error: aborting due to 1 previous error

